cpp/src/io/parquet/page_string_decode.cu

```diff
@@ -294,7 +296,6 @@ __device__ thrust::pair<int, int> page_bounds(page_state_s* const s,
       pp->num_nulls  = null_count;
       pp->num_valids = pp->num_input_values - null_count;
     }
-    __syncthreads();
```

Comment:
This seems dangerous to remove.  Aren't all threads except 0 in danger of using the wrong `pp->num_nulls` value?

Reply:
This is another only-valid-on-thread-0 result. I originally added _more_ syncthreads before all the other returns, but @vuule pointed out that once this function returns, all that happens is thread 0 takes the return values and copies them to global memory (along with 2 shared mem fields) and then returns. The other threads simply return ignored garbage and exit.